src/main/kotlin/app/revanced/patches/twitch/misc/settings/resource/patch/SettingsResourcePatch.kt

```diff
@@ -0,0 +1,62 @@
+package app.revanced.patches.twitch.misc.settings.resource.patch
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.ResourceContext
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patches.shared.settings.preference.impl.ArrayResource
+import app.revanced.patches.shared.settings.preference.impl.PreferenceScreen
+import app.revanced.patches.shared.settings.resource.patch.AbstractSettingsResourcePatch
+import app.revanced.patches.twitch.misc.settings.annotations.SettingsCompatibility
+
+@Name("settings-resource-patch")
+@SettingsCompatibility
+@Version("0.0.1")
+class SettingsResourcePatch : AbstractSettingsResourcePatch(
+"revanced_prefs",
+"twitch/settings"
+) {
+    override fun execute(context: ResourceContext): PatchResult {
+        super.execute(context)
+
+        /* required for self-restart in settings */
+        context.xmlEditor["AndroidManifest.xml"].use { editor ->
+            editor.file.getElementsByTagName("manifest").item(0).also {
+                it.appendChild(it.ownerDocument.createElement("uses-permission").also { element ->
+                    element.setAttribute("android:name", "android.permission.SCHEDULE_EXACT_ALARM")
+                })
+            }
+        }
```

Comment:
I have seen this code block for another patch as well, creating a patch and using it as a dependency on both, this and the other patch which uses the same code is probably a good idea to prevent code duplication, or you move it to `AbstractSettingsResourcePatch` into a method or the super method.